### PR TITLE
Details Navigation Adjustments

### PR DESCRIPTION
**Pull request by MatkoTom**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant
motivation and context. List any dependencies that are required for this change.

Updated the navigation logic when entering chats through the details screen.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration. All testing was done
manually by following, plainly, functionality requirements. If there are some special edge cases or
special ways that things need to be tested they will be mentioned below.

Dev testing

**Test Configuration**:

* Firmware version: A556BXXS3AXF2
* Hardware: SAMSUNG Galaxy A55
* SDK: Android 14

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

Media(2)
Media(3)
Media(4)

### CODE REVIEW

File: app/src/main/java/com/clover/studio/spikamessenger/ui/main/contact_details/ContactDetailsFragment.kt [hunk 1]
```diff
@@ -216,6 +219,22 @@ class ContactDetailsFragment : BaseFragment() {
                 )
             }
 
+            ivOpenChat.setOnClickListener {
+                user?.userId?.let { id ->
+                    run {
+                        CoroutineScope(Dispatchers.IO).launch {
+                            Timber.d("Checking room id: ${viewModel.checkIfUserInPrivateRoom(id)}")
+                            val roomId = viewModel.checkIfUserInPrivateRoom(id)
```
Comment (ivolarevic) [inline app/src/main/java/com/clover/studio/spikamessenger/ui/main/contact_details/ContactDetailsFragment.kt:227]: We can wrap this with let

File: app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat/ChatMessagesFragment.kt [hunk 2]
```diff
@@ -419,30 +419,31 @@ class ChatMessagesFragment : BaseFragment(), ServiceConnection {
...
-                    searchTerm: String?,
-                    selectedContentType: GPHContentType
-                ) {
-                    val gifUrl = media.images.original?.gifUrl.toString()
-                    handleGifClick(requireContext(), gifUrl)
-                }
-            }
-        }
+        // Gif view has been removed in version 1.2.1. It might return later.
+//        ivGif.setOnClickListener {
```
Comment (ivolarevic) [inline app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat/ChatMessagesFragment.kt:423]: Please return this xD

File: app/src/main/res/layout/fragment_chat_messages.xml [hunk 3]
```diff
@@ -148,17 +149,20 @@
                 android:background="@android:color/transparent"
                 android:src="@drawable/img_sticker"
                 app:layout_constraintBottom_toBottomOf="parent"
-                app:layout_constraintEnd_toStartOf="@id/iv_gif"
+                app:layout_constraintEnd_toEndOf="parent"
                 app:layout_constraintTop_toTopOf="parent" />
 
+            <!-- Gif view has been removed for now. It might return later.
             <ImageView
```
Comment (ivolarevic) [inline app/src/main/res/layout/fragment_chat_messages.xml:156]: Please return this

**Review by ivolarevic — APPROVED** [review]
LGTM

**Review by ivolarevic — APPROVED** [review]
LGTM
